src/lib/placeholders.ts

```diff
@@ -0,0 +1,36 @@
+import { getCountryName, getLocalCurrencySymbol } from './geolocation';
+
+const expectedPlaceholders = ['%%CURRENCY_SYMBOL%%', '%%ARTICLE_COUNT%%', '%%COUNTRY_NAME%%'];
+
+export const replacePlaceholders = (
+    content: string | undefined,
+    numArticles: number,
+    countryCode?: string,
+): string => {
+    if (!content) {
+        return '';
+    }
+
+    content = content.replace(/%%CURRENCY_SYMBOL%%/g, getLocalCurrencySymbol(countryCode));
+    content = content.replace(/%%ARTICLE_COUNT%%/g, numArticles.toString());
+
+    const countryName = getCountryName(countryCode) ?? '';
+    content = countryName ? content.replace(/%%COUNTRY_NAME%%/g, countryName) : content;
```

Comment:
so if `getCountryName(countryCode)` returns undefined then `countryName` is `''`.
And in that case,  the `%%COUNTRY_NAME%%` template will remain in the copy, which causes the `hasNoUnexpectedPlaceholders` to later fail?
Is the `?? ''` necessary?

Reply:
It's a good point. This is actually pre-existing code that I've simply moved into a separate file, but really we should unit test this bit too. I'll add some tests for this function to the PR. Thanks.